dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/utils/ParamUtils.java

```diff
@@ -95,6 +101,67 @@
         return globalParams;
     }
 
+    /**
+     * parameter conversion
+     * @param taskExecutionContext the context of this task instance
+     * @param parameters the parameters
+     * @return global params
+     */
+    public static Map<String,Property> convert(TaskExecutionContext taskExecutionContext, AbstractParameters parameters) {
+        Preconditions.checkNotNull(taskExecutionContext);
+        Preconditions.checkNotNull(parameters);
+        Map<String,Property> globalParams = getUserDefParamsMap(taskExecutionContext.getDefinedParams());
+        Map<String,String> globalParamsMap = taskExecutionContext.getDefinedParams();
+        CommandType commandType = CommandType.of(taskExecutionContext.getCmdTypeIfComplement());
+        Date scheduleTime = taskExecutionContext.getScheduleTime();
+
+        Map<String,Property> localParams = parameters.getLocalParametersMap();
+
+        if (globalParams == null && localParams == null) {
+            return null;
+        }
+        // if it is a complement,
+        // you need to pass in the task instance id to locate the time
+        // of the process instance complement
+        Map<String,String> timeParams = BusinessTimeUtils
+                .getBusinessTime(commandType,
+                        scheduleTime);
+
+        if (globalParamsMap != null) {
+            timeParams.putAll(globalParamsMap);
+        }
+
+        if (Strings.isNotBlank(taskExecutionContext.getExecutePath())) {
+            timeParams.put(Constants.PARAMETER_TASK_EXECUTE_PATH,taskExecutionContext.getExecutePath());
+        }
+        timeParams.put(Constants.PARAMETER_TASK_INSTANCE_ID,Integer.toString(taskExecutionContext.getTaskInstanceId()));
+
```

Comment:
1:There is an ambiguity in using timesParam to store temporary variables. timesParam mainly stores parameters of time type.

2:Is the result correct? Simply looking at the code, it does not seem to complete the final replacement.

3:If we want to support it, we suggest that other task types adopt the same approach to be consistent. (Maybe other tasks have the same need)

Reply:
1：For sure ,it's not pretty good way to use timesParam to store temporary variables, I will rename timesParam.

2: The result is correct . This feature has passed the unit test, and we have validated it in our environment. 
![image](https://user-images.githubusercontent.com/4928204/125011353-dfd81d80-e09a-11eb-8240-3a384974fb8b.png)

3：Sounds good. We will do that in next pr.

Reply:
It is best to create an issue to track this issue.

Reply:
hi,how it is getting?

Reply:
> hi,how is it getting?

First of all: create a new shell task, and then add this two parameters in the field of customer defined parameter. You can use these self-designed parameters in the script.

![image](https://user-images.githubusercontent.com/4928204/126157677-0a58111f-3f1a-4cf0-a15e-06cf56055264.png)

Second: save this task and run this flow to check this result.

![image](https://user-images.githubusercontent.com/4928204/126158315-1fea83a2-a5b2-4eec-8add-95f9f22c27f9.png)

Reply:
I am referring to legacy question.：）

Reply:
> I am referring to legacy question.：）

Got it. I've made a new issue for that. I'll implement this function to support the rest of type of tasks ASAP.

New issue is :https://github.com/apache/dolphinscheduler/issues/5852